library/core/src/num/int_macros.rs

```diff
@@ -511,6 +511,8 @@ macro_rules! int_impl {
                       without modifying the original"]
         #[inline(always)]
         #[cfg_attr(miri, track_caller)] // even without panics, this helps for Miri backtraces
+        #[requires(!self.overflowing_add(rhs).1)]
+        #[ensures(|ret| *ret >= $SelfT::MIN && *ret <= $SelfT::MAX)]
```

Comment:
I don't think you need an ensures here. This ensure clause is trivially true since all possible values are between MIN and MAX.

Reply:
That makes sense. I'll remove ensures in both files.